### ISSUE TITLE
Multiselection problem

### ISSUE THREAD

**Issue by jrscm** [punker76/gong-wpf-dragdrop#496]
Hello,

I am using this library (that I like a lot) with a DataGrid and I am experiencing a strange behaviour with multiselection, if I click on a selected item when more then one is selected, the selection change and only the item I clicked on remain selected.

Investigating in the code of DragDrop.cs I found this comment in the method DragSourceDown:
// If the sender is a list box that allows multiple selections, ensure that clicking on an  // already selected item does not change the selection, otherwise dragging multiple items  // is made impossible.
and the control is done with itemsControl.CanSelectMultipleItems() then in DragSourceUp there is condition to handle the multiselection
(Keyboard.Modifiers & ModifierKeys.Control) != 0 || itemsControl is ListBox listBox && listBox.SelectionMode == SelectionMode.Multiple
and since there is a check on the ListBox in my case with the DataGrid my multiselection is discarted.

I believe there is a mismatch in this two methods, either the multiselection is handled only for the ListBox or for every controls that satisfy CanSelectMultipleItems(), but the condition should be the same in both methods.
Anyway it is probably correct to handle it only for the ListBox since other ItemsControls could not need selection when items are clicked, in my case with the DataGrid I use checkboxes on row header and I don't want the selection to change by clicking on the DataGridCell of the items.

Does any of this make sense?
Let me know if something can be done to make this behaviour better.

Thanks!

**Comment by punker76** [comment]
@jrscm  I can not reproduce your issue

https://github.com/user-attachments/assets/529e86cf-db57-4fa9-b587-7d5011b328a2

**Comment by jrscm** [comment]
Hello,
maybe I didn't explain it well enough, the problem is not that the selection is lost when items are dragged, that works fine.

**Comment by jrscm** [comment]
Hello, maybe I didn't explain it well enough, the problem is not that the selection is lost when items are dragged, that works well.

**Comment by jrscm** [comment]
Hello,
maybe I didn't explain it well enough, the problem is not that the selection is lost when items are dragged, that works well.
The selection in a DataGrid can be more complex than in a ListBox, for example I could need to keep a row selected and at the same time click to select just one cell of another row, and as I tried to explain on the main post, the code will deselect the row that I have selected before and this is not a DataGrid standard behaviour, that's the problem that I am experiencing. This is a change in what happen in the control with just a single click, drag&drop is not even involved.

If it can help at the moment in my code I solved changing the check in DragSourceDown from:
` itemsControl.CanSelectMultipleItems()`
to:
`sender is ListBox listBox && listBox.SelectionMode == SelectionMode.Multiple`
as is in DragSourceUp, in this way the DataGrid behaves as expected when a single cell is clicked and the other rows selection is preserved.
